From ad817f52659d199499faa3a0e22a7ddf9959f698 Mon Sep 17 00:00:00 2001
From: rmasarovic <r.masarovic@gmail.com>
Date: Sun, 17 Apr 2022 21:40:51 +0200
Subject: [PATCH 1/2] fix growable list

---
 lib/src/Puzzle.dart | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/src/Puzzle.dart b/lib/src/Puzzle.dart
index 7da0b2b..2ce4762 100644
--- a/lib/src/Puzzle.dart
+++ b/lib/src/Puzzle.dart
@@ -121,7 +121,7 @@ class Puzzle {
     Cell _target = _board!.cellAt(position);
     _target.setValue(value);
 
-    List<CellViolation> _violations = new List<CellViolation>.empty();
+    List<CellViolation> _violations = new List<CellViolation>.empty(growable: true);
 
     if (board()!.isRowViolated(position)) {
       _violations.add(CellViolation.Row);

From ae28cdbb89335d716d264e8bc5f4d784b305abe1 Mon Sep 17 00:00:00 2001
From: rmasarovic <r.masarovic@gmail.com>
Date: Mon, 2 May 2022 02:41:39 +0200
Subject: [PATCH 2/2] fix error pattern set validator

---
 lib/src/models/PatternSet.dart |  4 ++--
 test/pattern_set_test.dart     | 14 ++++++++++++++
 2 files changed, 16 insertions(+), 2 deletions(-)
 create mode 100644 test/pattern_set_test.dart

diff --git a/lib/src/models/PatternSet.dart b/lib/src/models/PatternSet.dart
index fdc19c0..da0f925 100644
--- a/lib/src/models/PatternSet.dart
+++ b/lib/src/models/PatternSet.dart
@@ -118,10 +118,10 @@ class PatternSet<Pattern> extends DelegatingList<MY.Pattern> {
   Tuple2<bool, List<String>> _checkValidPatternMap(Map<int, String>? map) {
     String _patrow = "";
     String _bigrow = "";
-    List<String> _reasons = new List<String>.empty();
+    List<String> _reasons = new List<String>.empty(growable: true);
 
     for (int row = 0; row < 9; row++) {
-      _patrow = map![row]!.replaceAll(' ', '');
+      _patrow = map![row]?.replaceAll(' ', '')??"0";
 
       if (_patrow.length != 9) {
         _reasons.add("Pattern Map row #$row does not " +
diff --git a/test/pattern_set_test.dart b/test/pattern_set_test.dart
new file mode 100644
index 0000000..8944c30
--- /dev/null
+++ b/test/pattern_set_test.dart
@@ -0,0 +1,14 @@
+import 'package:sudoku_api/src/logic/SudokuException.dart';
+import 'package:test/test.dart';
+import 'package:sudoku_api/sudoku_api.dart';
+
+void main() async {
+  String patterSetTest = 'Error Pattern Set';
+  test(patterSetTest, () {
+    final Map<int, String> errorMap = {0: "0 0 0 0 1 0 0 0 0 1 0",};
+    final patterSet= PatternSet();
+    final patterName="errorPattern";
+    Pattern pattern= Pattern(patterName, errorMap);
+    expect(() => patterSet.add(pattern), throwsA(isA<InvalidPatternException>()));
+  });
+}
\ No newline at end of file
